crux_http/src/lib.rs

```diff
@@ -26,7 +26,7 @@ pub use http_types::{self as http};
 
 pub use self::{
     config::Config,
-    error::HttpError,
+    error::{HttpError, HttpErrorHttp},
```

Comment:
Doesn't look like it was introduced by this PR but `HttpErrorHttp` seems like a bad name for a type....

Reply:
We could make that into a struct variant on the `HttpError` and move the impl into `HttpError` I suppose 🤔 That's why it's called it - it's a struct for the `Http` variant of the `HttpError`

Reply:
Hmm, yes, I see.  Well 🤷 I have no real preference, just saw the name and thought "what is going on". 

Will leave yous to decide whether or not it's fine.

Reply:
I made it a struct variant. Seems much cleaner